### DISCUSSION TITLE
Using GPIO inputs as interrupts

### DISCUSSION THREAD

**Discussion by thedim-witted** [eblot/pyftdi#294] [Q&A]
Hi,

I wanted to know if we have a function through which we can attach a GPIO input as an interrupt to my system, without having the need to continuously monitor the state?

**Comment by eblot** [comment]
Unfortunately the FTDI device does not have a USB interrupt channel, so the host has to poll the FTDI device to get an updated status. Note that even an interrupt channel this would also be the case, but that would have been handled by the host's USB stack, not the application.

**Reply by thedim-witted** [reply]
Is there a way to tie the FTDI input to the interrupt channel of the OS?

**Reply by eblot** [reply]
I'm not sure what you meant. The short answer would be no. There is no "interrupt channel of the OS"; by interrupt channel, I meant a USB interrupt endpoint, sorry for the confusion.

**Reply by thedim-witted** [reply]
I was going through the pyusb library and it seems that it has the capability to connect the usb interrupt endpoint to any device connected via the usb.

**Reply by eblot** [reply]
I think you do not get it. It is not a SW limitation, it is a HW limitation. FTDI devices do not have a USB interrupt endpoint, only bulk endpoints.

**Reply by thedim-witted** [reply]
Oh ok. That is a bummer.
